demo_agent/agent.py

```diff
@@ -322,7 +327,7 @@ def get_action(self, obs: dict) -> tuple[str, dict]:
                 {"role": "user", "content": user_msgs},
             ],
         )
-        action = response.choices[0].message.content
+        action = extract_code_blocks(response.choices[0].message.content)[1]
```

Comment:
Have you tried this? The new `extract_code_blocks()` function returns a list of tuples, but action is expected to be a string